### ISSUE TITLE
Permission for LZX decompressor code in project?

### ISSUE THREAD

**Issue by Mgamerz** [gildor2/UEViewer#170]
Hi Gildor,

I've been doing a lot of data mining from xbox 360 games and made a wrapper dll project to allow 3rd party apps to decompress LZX data. It's based on code from UEViewer, though about 90% of it is from MSPack which is LGPL according to it's files. The repo is here for you to peruse https://github.com/ME3Tweaks/lzxdhelper. I have a wrapper function that allows me to call native LZX decompression easily with C# but it will work with anything that can interface with native.

I traced how your app did LZX decompression then pulled the relevant code into lzxmain.h. Since this repo doesn't have a license, can I have permission to redistribute this small block of code of yours? I'm still setting up the project so I can add permissions and other files as necessary.

**Comment by gildor2** [comment]
Hi,

I'm ok with reusing my code. I have plans adding BSD license for my repo, not sure how I could use LGPL mspack after that (when I added it, I didn't bother much about licenses) - probably I'll put a reference to the repository and make compilation conditional (like I did with ooz).

Take a note that I've made some fixes to mspack because it was failed to decompress some XBox360 (XMemCompress) blocks.

It would be nice to have a reference to original code or just to UModel in your code or readme, but I'm ok if you won't do that.

Konstantin
